### PR TITLE
SwiftSyntax: reduce the code size by adding fixed layout attr.

### PR DESCRIPTION
**Pull request by shajrawi**
radar rdar://problem/38581539

### CODE REVIEW

File: tools/SwiftSyntax/SwiftcInvocation.swift [hunk 1]
```diff
@@ -22,27 +22,35 @@ import Glibc
 
 /// The result of process execution, containing the exit status code,
 /// stdout, and stderr
+@_versioned
+@_fixed_layout
 struct ProcessResult {
   /// The process exit code. A non-zero exit code usually indicates failure.
+  @_versioned
```
Comment (slavapestov) [inline tools/SwiftSyntax/SwiftcInvocation.swift]: @_versioned is unnecessary here or anywhere else

**Comment by shajrawi** [comment]
@nkcsgexi can you please review? this reduces the library size with resilience enabled by slightly over 0.5MB

**Review by slavapestov — CHANGES_REQUESTED** [review]
Does code size for SwiftSyntax really matter? We're missing out on testing resilience by marking everything as fixed layout.

We could just disable resilience entirely when building this library too.

**Comment by slavapestov** [comment]
@shajrawi Alternatively, we should instead spend time figuring out why the code size increased instead of just disabling it.

**Comment by nkcsgexi** [comment]
@shajrawi thanks for doing this! i guess reduced code size is a good thing no matter what 😄

**Review by nkcsgexi — APPROVED** [review]

**Comment by harlanhaskins** [comment]
I don't think code size matters much for SwiftSyntax because it only lives on host devices (it's never shipped on iOS, so there's only the one copy per toolchain).

**Comment by shajrawi** [comment]
Closing this PR - per the offline discussion.

**Pull request closed without merging**